dask/bag/random.py

```diff
@@ -82,6 +82,7 @@ def _sample_map_partitions(population, k, replace):
     k: int
         Number of elements to sample.
     """
+    population = list(population)
```

Comment:
@jrbourbeau since you know much more about bags then me can you check that this is an ok thing to do?

Reply:
Given that we're calling `len` I'm not sure if there is any other choice here, unless we want to become fancy.

Most bag operations actually support out-of-memory partitions.  You could have a bag with a single partition over a terabyte of data, and most operations would actually be ok.  For sampling, if we wanted to continue supporting out-of-memory operations, we would probably implement [reservoir sampling](https://en.wikipedia.org/wiki/Reservoir_sampling).

Short term, I recommend that we merge this (it's certainly an improvement over things breaking before), and then raise an issue about reservoir sampling.  I think that it would be a good second issue for someone.  It's the kind of thing that someone with some CS background will enjoy working on, even if they don't know a lot about Dask itself.

Reply:
Ok thanks for the feedback. I opened https://github.com/dask/dask/issues/7068